### PR TITLE
JCRVLT-438 add RCP API

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: vault-rcp/src/main/java/org/apache/jackrabbit/vault/rcp/impl/RcpTaskImpl.java [hunk 1]
```diff
@@ -211,18 +210,22 @@ public STATE getState() {
...
     }
 
+    @Override
     public String getDestination() {
         return dst;
     }
 
+    @Override
     public void setRecursive(boolean b) {
         this.recursive = b;
```
Comment (tripodsan) [inline vault-rcp/src/main/java/org/apache/jackrabbit/vault/rcp/impl/RcpTaskImpl.java:225]: don't allow modification if the task is running.

File: vault-rcp/src/main/java/org/apache/jackrabbit/vault/rcp/impl/RcpTaskImpl.java [hunk 2]
```diff
@@ -211,18 +210,22 @@ public STATE getState() {
...
     }
 
+    @Override
     public void addExclude(String exclude) throws ConfigurationException {
```
Comment (tripodsan) [inline vault-rcp/src/main/java/org/apache/jackrabbit/vault/rcp/impl/RcpTaskImpl.java:229]: don't allow modification if the task is running.

File: vault-rcp/src/main/java/org/apache/jackrabbit/vault/rcp/impl/RcpTaskImpl.java (hunk 1 above)
Comment (kwin) [reply vault-rcp/src/main/java/org/apache/jackrabbit/vault/rcp/impl/RcpTaskImpl.java:225]: I will make recursive final

File: vault-rcp/src/main/java/org/apache/jackrabbit/vault/rcp/impl/RcpTaskImpl.java (hunk 2 above)
Comment (kwin) [reply vault-rcp/src/main/java/org/apache/jackrabbit/vault/rcp/impl/RcpTaskImpl.java:229]: I will make the filter final

**Pull request merged**